Source/Logger.swift

```diff
@@ -83,6 +98,44 @@ open class Logger {
         self.executionMethod = executionMethod
     }
 
+    // MARK: -  Filtering & changing log levels
+
+    /// Sets a new log level on the logger. Any previously logged messages will be emitted based on the setting
+    /// at the time they were logged.
+    /// - Parameter level: The new minimum log level
+    public func setLogLevels(_ levels: LogLevel) {
+        executionMethod.perform {
+            // if this is an async serial queue, this work we are in will happen _after_
+            // any of the previously enqueued log messages are written. Therefore, this
+            // ensures that messages enqueued after this call will be using the new log level
+            // filter.
+            self.logLevels = levels
+        }
+    }
+
+    /// Adds a log filter to the logger. A filter gives you dynamic control over whether logs are emitted or not, based on content in the ``LogMessage`` struct or message itself.
+    /// - Parameter filter: A ``LogFilter`` instance to add.
+    public func addFilter(_ filter: LogFilter) {
```

Comment:
Good choice in naming by dropping `Log` in the naming convention to match how we refer to writers.